### ISSUE TITLE
Allow product lines to be left partially filled in

### ISSUE THREAD

**Issue by jkupczak** [hql287/Manta#97]
Please answer these question so I can help you faster.

**Is it a bug or a feature request?**

- [ ] It's a bug.
- [x] I'm requesting a new feature.

**Which version are you using?**

- [x] macOS
- [ ] Windows
- [ ] Linux

**Which is the version/distribution of the OS?**

OSX El Capitan

**Describe your issue**

I can't save an invoice when a product line is empty.

**Steps to replicate**

Fill in the first product line that is provided with a description, price, and quantity. Then hit enter and a new line will be added. Try to save.

---

I have a habit of hitting enter after I'm done with a line and that ends up creating a new line. Which is perfect. But I feel as though there shouldn't be a block on saving and previewing if a line item is left completely empty.  And in fact, I could find use for line items that only included 1 or 2 of the fields filled in.

If a line is completely empty, instead of requiring me to hit the X to remove it, just ignore it. And if I only enter some of the data, just use that instead of requiring me to finish it.

**Comment by edmael** [comment]
Hi @hql287 
I'm trying to contribute to your project by adding this feature, but I'm having some trouble.

Here's the repository with the changes: https://github.com/edmael/Manta

Basically I edited these three files:
app.js - added the enableEmptyRow variable
forms.js - added an if statement to avoid the check if enableEmptyRow set to true
General.jsx - created a switch for enableEmptyRow true/false

Can you help me troubleshoot it?
What am I missing?

Thanks, have a nice day.

**Comment by hql287** [comment]
Hi Edoardo, thanks a lot for being interested in contributing to this project. 👍

Could you make a PR so we can work on it together? 

Thanks!

**Comment by hql287** [comment]
> And if I only enter some of the data, just use that instead of requiring me to finish it.

@jkupczak: Can you be more specific? It wouldn't make sense if you just put in the name of the product/service without its respective quantity and price.

**Comment by edmael** [comment]
@hql287 I think some people may be using Manta not just for the invoices, but also for quotations or something like that.

Even if you don't consider that and want your application used only for the invoices, there may be cases where someone wants to add a "non _concrete_ product" (like, "working hours" or something like that - I reckon it's not the best example, I know) and leave quantity and price aside.

**Comment by hql287** [comment]
> I think some people may be using Manta not just for the invoices, but also for quotations or something like that.

Well, then we should develop a new feature to facilitate this instead of removing/disabling the current validation. In fact, someone already requested this feature https://github.com/hql287/Manta/issues/75. 

I think the best way to go about this would be adding a drop-down or radio buttons for the user to choose whether they want to make an estimate/quotation/invoice/receipt. Based on this value we can use an appropriate validation. 

Let me know if you want to work on this, @edmael 👍

**Comment by edmael** [comment]
@hql287 yeah, the idea was to add a switch (we can call it "Enable Empty Line" or "Enable quotation") that allows the users to bypass the row check.

In my pull request I tried to do that: add a switch in the settings page that, if enabled, disable the check of empty rows.

The best way to do this, however, is like you said: a drop down menu in the settings (or in the invoice page) that allows the users to choose what kind of document they want to generate. If it's just a quotation or something like that we disable the empty row check.

**Comment by hql287** [comment]
@edmael: I'm glad we came to the same conclusion. Can you confirm that you're already working on my suggestion? If so, please make a separate PR or edit https://github.com/hql287/Manta/pull/186 so if reflects correctly what you're doing. 

Maybe smt like 'Allow creating Esitmate/Quotation"? This would be a much bigger PR though. Let me know.

Thanks!

**Comment by edmael** [comment]
Yeah, it's going to be a bigger work but I'm still willing to do it.

Not sure I **can** do it (not a guru of electron and js), but I'll definitely try and do a PR as soon as I have something.

**Comment by hql287** [comment]
> Not sure I can do it (not a guru of electron and js), but I'll definitely try and do a PR as soon as I have something.

Yeah, just give it a go and let me know if I can help with anything. Thanks!

**Comment by jkupczak** [comment]
> @jkupczak: Can you be more specific? It wouldn't make sense if you just put in the name of the product/service without its respective quantity and price.

@edmael suggested I may be using Manta for quotations. And while that is true, my comment about leaving off price and quantity wasn't related to that. 

In the case of us selling a software service as a package, we have a need to add in separate features as their own line items. These various features do not have a specific price attached to them. At least not one we want to show the customer. But we do want them to be displayed as their own items.

An extension of this is that it would be great to be able to list a line item not only with a blank price and quantity, but also with a price of $0 or simply with text like "Free".

**Comment by hql287** [comment]
> An extension of this is that it would be great to be able to list a line item not only with a blank price and quantity, but alternatively with a price of $0 or simply with text like "Free".

This is interesting. It's a perfectly valid use case and I've never thought about this. I think I understand what you meant now and I can imagine that you're on alone on this. 

@edmael any update on this? Do you think we should make a separate PR for this case? Or do you want to tackle it with your current PR?
